src/x86/abm.rs

```diff
@@ -0,0 +1,71 @@
+//! Advanced Bit Manipulation (ABM) instructions
+//! 
+//! That is, POPCNT and LZCNT. These instructions have their own CPUID bits to
+//! indicate support.
+//!
+//! TODO: it is unclear which target feature to use here. SSE4.2 should be good
+//! enough but we might need to use BMI for LZCNT if there are any problems.
+//!
+//! The references are:
+//!
+//! - [Intel 64 and IA-32 Architectures Software Developer's Manual Volume 2: Instruction Set Reference, A-Z](http://www.intel.de/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-instruction-set-reference-manual-325383.pdf).
+//! - [AMD64 Architecture Programmer's Manual, Volume 3: General-Purpose and System Instructions](http://support.amd.com/TechDocs/24594.pdf).
+//!
+//! [Wikipedia](https://en.wikipedia.org/wiki/Bit_Manipulation_Instruction_Sets#ABM_.28Advanced_Bit_Manipulation.29)
+//! provides a quick overview of the instructions available.
+
+
+/// Counts the leading most significant zero bits.
+///
+/// When the operand is zero, it returns its size in bits.
+#[inline(always)]
+#[target_feature = "+sse4.2"]
```

Comment:
Clang seems to use the `lzcnt` target feature: https://github.com/llvm-mirror/clang/blob/a38ba9770a70056f9fdd6c71f819e5db45a105e4/lib/Headers/lzcntintrin.h

Should we be doing the same?

Reply:
@BurntSushi I have a comment on the top of the module with TODO mentioning this. IIRC  `lzcnt` is not a feature that `cfg(target_feature)` supports.  

That is, we would need to add support for `lzcnt` there, and in the future run-time detection support. The implementation cost of doing this is small, but all future code that wants to be portable would need to distinguish lzcnt from popcnt and SSE4.2. Whether this is worth it, I don't know, but I'd like to fill an issue in this repo to discuss it.

Reply:
@BurntSushi I tried to find an alternative, but I couldn't, so I've fixed this.